Some Notes
-----------

* There currently exist a pleathora of methods which can be used to obtain clustering
	* Some are generalized ML methods
	* Some are 'specialized' for single cell (although I would argue many are just a rebranding of existing techniques)

* Clustering algorithms tend to be anti-conservative
	* They are very good at detecting patterns that do exist
	* However, they have a tendancy to also detect patterns that don't exist

* Each algorithm will have it's own tendancies and biases
	* One method to help identify these is to use internal validation measures
		* Compactness
		* Connectedness
		* Spatial Separation
		* Combinations
			* Dunn Index (Dunn 1974)
			* Davids-Bouldin Index (1979)
			* Silhouette Width (Rousseeuw 1987)
		* Predictive power/stability
			* Ben-Hurr 2002
			* Bittner 2000
			* Brechenridge 1989
			* Fridlyand & Dudoit 2001
			* Kerr and Churchill 2001
			* Lange 2004
			* Levine & Domany 2001
			* Li and Wong 2001
			* McShane 2002
			* Tibshirani 2001 **
		* "Compliance between partitioning and distance information"
			* Romesburg 1984

* Handl 2005 provides some advice for performing clustering analyses
	* Use multiple clustering algorithms/approaches
		* If a cluster persistently exists between algorithms with different clustering tendancies, it may be more likely to exist
		* Hence applying a consensus algorithm may provide more realistic results

Practical Approach/Ideas
------------------------
* Before Clustering, we should generate one of the many statistics that measure clustering tendancy.
	* [Hopkins Statistic](https://en.wikipedia.org/wiki/Hopkins_statistic)
* We apply a variety of [clustering algorithms](https://scikit-learn.org/stable/modules/clustering.html)
* We can also experiment with different dimensionality reduction techniques
	* UMAP
	* t-SNE (cannot be applied to new data)
	* SDAE
	* ... ?
* Compare each of the resulting clusterings
	* [Adjusted Rand Index](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.adjusted_rand_score.html) (ARI)
	* [Normalized Mutual Information](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.normalized_mutual_info_score.html) (NMI ... There is also a version that is adjusted for chance similar to ARI)
	* Visualization
		* We can further reduce the number of dimensions to 2, and color using the results of various clustering algorithms to show similarities/differences
		
Techniques for Consensus Clustering?
------------------------------------
* [Wikipedia on Consensus Clustering](https://en.wikipedia.org/wiki/Consensus_clustering)
	* Hard Consensus
		* Cluster-Based Similarity Partitioning Algorithm (CSPA)
		* Hyper-Graph Partitioning Algorithm (HGPA)
		* Meta-Clustering Algorithm (MCLA)
	* Soft Consensus
		* sCSPA
		* sMCLA
		* sHGPF
		* Bayesian Consensus Clustering

Remaining Issues
----------------
* What clustering algorithms to use
* What dimensionality reduction techniques to use
	* How do different dimensionality techniques affect clustering
* How many dimensions to reduce to
	* How does a different # of dimensions affect clustering

Note: Paukkeri 2011 (Effect of Dimensionality Reduction on Different Distance Measures in Document Clustering) may be of interest

Helpful Papers
--------------
### Clustering Methodology
* [Computational cluster validation in post-genomic data analysis](https://academic.oup.com/bioinformatics/article/21/15/3201/195678)
* [Effect of Dimensionality Reduction on Different Distance Measures in Document Clustering](https://link.springer.com/chapter/10.1007/978-3-642-24965-5_19)

### Clustering Statistics
* [Validating clustering for gene expression data](https://academic.oup.com/bioinformatics/article/17/4/309/328189)
	** Introduces 'Figure of Merit'
### Consensus Clustering
* [Critical limitations of consensus clustering in class discovery](https://www.nature.com/articles/srep06207)
* [A reassessment of consensus clustering for class discovery](https://www.biorxiv.org/content/biorxiv/early/2014/02/13/002642.full.pdf)

## Applications in single-cell data
* [SAFE-clustering: Single-cell Aggregated (From Ensemble) Clustering for Single-cell RNA-seq Data](https://www.biorxiv.org/content/early/2018/08/06/215723)
* [A systematic performance evaluation of clustering methods for single-cell RNA-seq data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6134335/)
	** Makes reference to 12 datasets that may be useful for benchmarking
* [scRNA-seq mixology: towards better benchmarking of single cell RNA-seq protocols and analysis methods](https://www.biorxiv.org/content/early/2018/10/08/433102)
* [Identification of cancer subtypes from single-cell RNA-seq data using a consensus clustering method](https://bmcmedgenomics.biomedcentral.com/articles/10.1186/s12920-018-0433-z)

One last note
-------------
I wasn't sure how to work this in, but I found it interesting so I thought I might include/document it.
A few days back I created embeddings of the data reduced down to 2 dimensions using a few techniques.
DBSCAN can then be used to obtain a clustering for each embedding.  I was curious to visualize the differences
between approaches, so I used the clustering of one approach in order to colour the embedding on another.

There are a few more, but here is one:
![Umap clusters shown on pca->tsne embedding]()

